src/app.js

```diff
@@ -0,0 +1,275 @@
+      formatNameAttr =
+        'FFVideoFormat1080p' +
+        this.#mediaInfo.frameRate
+          .toFixed(2)
+          .toString()
+          .replace('.', '')
+    }
+    let formatFrameDurationAttr = this.toTimecodeStr(1) // 1/frameRate
+    let formatWidthAttr = this.#mediaInfo.width
+    let formatHeightAttr = this.#mediaInfo.height
+    if (this.#mediaInfo.hasVideo) {
+      this.#startStr = `<?xml version="1.0" encoding="UTF-8"?><!DOCTYPE fcpxml><fcpxml version="1.9"><resources><format id="r0" name="${formatNameAttr}" width="1920" height="1080" frameDuration="${formatFrameDurationAttr}"/><format id="r1" name="FFVideoFormatRateUndefined" width="${formatWidthAttr}" height="${formatHeightAttr}" frameDuration="${formatFrameDurationAttr}"/>`
+      this.#endStr = `</spine></sequence></project></event></library></fcpxml>`
+      this.#currentAssetID = 2 // 0 and 1 are reserved for <format>
+    } else {
+      this.#startStr = `<?xml version="1.0" encoding="UTF-8"?><!DOCTYPE fcpxml><fcpxml version="1.9"><resources><format id="r0" name="FFVideoFormat1080p24" width="1920" height="1080" frameDuration="1/24s"/>`
+      this.#endStr = `</gap></spine></sequence></project></event></library></fcpxml>`
+      this.#currentAssetID = 1 // 0 is reserved for <format>
+    }
+    this.#assetList = []
+    this.#eventStr = `</resources><library><event name="Timeline 1"><project name="Timeline 1">`
+    this.#sequenceStr = ''
+    this.#clipList = []
+    this.#currentOffset = this.#initialOffset
+  }
+  gcd (a, b) {
+    let temp
+    while (b !== 0) {
+      temp = a % b
+      a = b
+      b = temp
+    }
+    return a
+  }
+  toTimecodeStr (frameNumber) {
+    let tcNumerator = frameNumber * this.#tcMultiplier
+    let divisor = this.gcd(tcNumerator, this.#tcDenominator)
+    return (
+      (tcNumerator / divisor).toString() +
+      '/' +
+      (this.#tcDenominator / divisor).toString() +
+      's'
+    )
+  }
+  addClip (startFrame, endFrame) {
+    let asset = ''
+    let clip = ''
+    let assetIDAttr = 'r' + this.#currentAssetID.toString()
+    let assetNameAttr = this.#mediaInfo.fileName
+    let assetDurationAttr = this.toTimecodeStr(this.#mediaInfo.frameCount)
+    let clipStartAttr = this.toTimecodeStr(startFrame)
+    let clipDuration = endFrame - startFrame
+    let clipDurationAttr = this.toTimecodeStr(clipDuration)
+    let clipOffsetAttr = this.toTimecodeStr(this.#currentOffset)
+    if (this.#mediaInfo.hasVideo) {
+      asset = `<asset id="${assetIDAttr}" format="r1" name="${assetNameAttr}" start="0/1s" duration="${assetDurationAttr}" hasVideo="1" hasAudio="1" audioSources="1"><media-rep kind="original-media" src="${assetNameAttr}"/></asset>`
+      clip = `<asset-clip ref="${assetIDAttr}" format="r1" tcFormat="NDF" name="${assetNameAttr}" start="${clipStartAttr}" duration="${clipDurationAttr}" offset="${clipOffsetAttr}" enabled="1"><adjust-transform scale="1 1" anchor="0 0" position="0 0"/></asset-clip>`
```

Comment:
I don't think adjust transform is needed. It is in stock da Vinci da Vinci processes file without adjust transform just fine

Reply:
Yeah, there is definitely room for simplification- let's start testing what attributes and tags we can safely remove after this gets merged, and remove everything that's not needed.